lst_scripts/onsite_mc_r0_to_dl1.py

```diff
@@ -115,66 +162,123 @@ def main():
-            cmd = 'sbatch -e {} -o {} {} {}'.format(jobe, jobo, base_cmd, os.path.join(dir_lists, file))
 
-            # os.system(cmd)
-            print(cmd)
+            # recover or not the jobid depending of the workflow mode
+            if not flag_full_workflow:
+                cmd = 'sbatch -e {} -o {} {} {}'.format(jobe, jobo, base_cmd, os.path.join(dir_lists, file))
+
+                print(cmd)
+                # os.system(cmd)
+
+            else:  # flag_full_workflow == True !
+                job_name = {'electron': 'r0dl1_e',
+                            'gamma': 'r0dl1_g',
+                            'gamma-diffuse': 'r0dl1_gd',
+                            'proton': 'r0dl1_p'
+                            }
+
+                particle_type = DL0_DATA_DIR.split('/')[-2]
+
+                cmd = 'sbatch --parsable -J {} -e {} -o {} {} {}'.format(job_name[particle_type],
+                                                                         jobe, jobo,
+                                                                         base_cmd, os.path.join(dir_lists, file))
+
+                jobid = os.popen(cmd).read().strip('\n')
+                jobids_r0_dl1.append(jobid)
+
+                # Fill the dictionaries if IN workflow mode
+                jobid2log[jobid] = {}
+                jobid2log[jobid]['particle'] = particle_type
+                jobid2log[jobid]['set_type'] = set_type
+                jobid2log[jobid]['jobe_path'] = jobe
+                jobid2log[jobid]['jobo_path'] = jobo
+                jobid2log[jobid]['sbatch_command'] = cmd
+
+                # print(f'\t\t{cmd}')
+                print(f'\t\tSubmitted batch job {jobid}')
+
             counter += 1
 
-        print("{} jobs submitted".format(counter))
+        print("\n\t{} jobs submitted".format(counter))
 
     # copy this script itself into logs
     shutil.copyfile(sys.argv[0], os.path.join(RUNNING_DIR, os.path.basename(sys.argv[0])))
     # copy config file into logs
-    if args.config_file is not None:
-        shutil.copy(args.config_file, os.path.join(RUNNING_DIR, os.path.basename(args.config_file)))
+    if config_file is not None:
+        shutil.copy(config_file, os.path.join(RUNNING_DIR, os.path.basename(config_file)))
 
     # save file lists into logs
     shutil.move('testing.list', os.path.join(RUNNING_DIR, 'testing.list'))
     shutil.move('training.list', os.path.join(RUNNING_DIR, 'training.list'))
 
-    print("\n ==== END {} ==== \n".format(sys.argv[0]))
+    # create log dictionary and return it if IN workflow mode
+    if flag_full_workflow:
 
+        print("\n ==== END {} ==== \n".format('r0_to_dl1_workflow'))
+        return jobid2log, jobids_r0_dl1
 
+    else:
+        print("\n ==== END {} ==== \n".format(sys.argv[0]))
 
-if __name__ == '__main__':
-    main()
 
+if __name__ == '__main__':
+    args = parser.parse_args()
+    main(args.input_dir,
```

Comment:
how can user  set `flag_full_workflow` ?

Reply:
`flag_full_workflow` is supposed to be set modified ONLY at the `r0_to_dl3` script. Why ? Because by default is set to `False` - what allows each single stage to be run separately -, and this way avoids the user to take into account all the rest of `fla_full_workflow = True` dependent arguments - what simplifies the run of each stage separately.

Reply:
got it thanks